crates/analytics/src/payment_intents/core.rs

```diff
@@ -256,10 +256,20 @@ pub async fn get_metrics(
             if let Some(retried_amount) = collected_values.smart_retried_amount {
                 let amount_in_usd = id
```

Comment:
You can do something like this:

```rust
let amount_in_usd = id
    .currency
    .and_then(|currency| {
        i64::try_from(retried_amount)
            // Log error (and return original result).
            // Passing `?error` to the logger so that it is included as a separate field in the log line.
            .inspect_err(|error| logger::error!(?error, "Amount conversion error"))
            .and_then(|amount_i64| {
                convert(ex_rates, currency, Currency::USD, amount_i64)
                    .inspect_err(|error| logger::error!(?error, "Currency conversion error")) // Log error (and return original result)
            })
            .ok() // Convert Result to Option
    })
    .map(|amount| (amount * rust_decimal::Decimal::new(100, 0)).to_u64())
    .unwrap_or_default();
```

Please remove the comments, I added them only for your understanding.